src/box/lua/schema.lua

```diff
@@ -241,6 +241,14 @@ local function revoke_object_privs(object_type, object_id)
     end
 end
 
+-- Check if schema version matches Tarantool version and raise an error if not.
+local function check_schema_version(options)
+    local disable_check = options and options._force
+    if internal.schema_needs_upgrade() and not disable_check then
+        box.error(box.error.DDL_NOT_ALLOWED, "box.schema.upgrade() is required")
```

Comment:
Let's make the error more verbose. For example, look at
https://github.com/tarantool/tarantool/blob/6e45042473fb973c7b1ad87fcf04c9081a4c40fa/src/box/lua/load_cfg.lua#L979-L990

Reply:
Done